shared/src/main/scala/tastyquery/ast/Types.scala

```diff
@@ -1249,6 +1281,12 @@ object Types {
       bounds.high.findMember(name, pre)
   }
 
+  enum TypeMemberDefinition:
```

Comment:
these don't need to be in types

Reply:
I moved it to `Symbols`.

I also added a big documentation comment on `Symbols`, and expanded the commit message.